src/lib/gssapi/generic/gssapi_ext.h

```diff
@@ -571,6 +571,31 @@ gss_import_cred(
     gss_buffer_t,              /* token */
     gss_cred_id_t *);          /* cred_handle */
 
+
+/*
+ * Channel Binding Extensions
+ * https://tools.ietf.org/html/draft-ietf-kitten-channel-bound-flag-01
+ */
+
+#define STUB_MAGIC_ID 0x00000ca7
+
+#define GSSINT_CHK_STUB(p) (!GSSINT_CHK_LOOP(p) && \
+    p->initial_ctx_id != NULL && \
+    ((stub_gss_ctx_id_rec *)p->initial_ctx_id)->magic_num == STUB_MAGIC_ID)
+
+
+typedef struct _stub_gss_ctx_id_rec {
```

Comment:
Also, I don't think a new structure was needed, just a flag in the existing context structures (union and for each provider).